src/sys/reboot.rs

```diff
@@ -0,0 +1,50 @@
+//! Reboot/shutdown or enable/disable Ctrl-Alt-Delete.
+
+use {Errno, Error, Result};
+use libc;
+use void::Void;
+use std::mem::drop;
+
+/// How exactly should the system be rebooted.
+///
+/// See [`set_cad_enabled()`](fn.set_cad_enabled.html) for
+/// enabling/disabling Ctrl-Alt-Delete.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum RebootMode {
```

Comment:
Having looked at your commit history, your problem probably was that you still have to cast the enumeration elements to `libc::c_int` in order to pass them to `reboot`, i.e.

```
reboot(cmd as libc::c_int)
```